wireless_modules/wireless_module.py

```diff
@@ -38,6 +42,22 @@ def __init__(self, module_id, battery_reader=None):
         self.mqtt = Client(client_id, config.MQTT_BROKER)
 
         self.battery = battery_reader
+        self.status_led = status_led
+
+        asyncio.get_event_loop().set_exception_handler(self.error_handler)
+
+    def error_handler(self, _loop, context):
+        self.status_led.set_state(WmState.Error)
+        print("An error occured in a uasyncio task!")
+        print(context["message"])
+        sys.print_exception(context["exception"])
+
+        # FIXME: A nicer solution would be to restart all running tasks rather than reset
+
+        print("Resetting in 5 seconds...")
+
+        time.sleep(5)
+        machine.reset()
```

Comment:
We already catch exceptions and do this in `back_module/main.py`, if errors thrown by async functions reach that `try and catch` block, we can just rely on that to reset the machine.

Reply:
Good catch, but actually asyncio catches and suppresses the errors, even though it may kill the tasks. It does print the traceback by default, but won't trigger a reset meaning the WM is out of action until manually power cycled.

Reply:
So does that mean the try catch block is redundant in the `main.py` script?

Reply:
The try catch there is still useful in case the initialisation crashes, that won't be caught by asyncio.